server/level_loader.cpp

```diff
@@ -23,69 +23,67 @@ void initPlayerwall(const FlatBuffGenerated::PlayerWall *pw)
-	myBodyDef.position.Set(dfstone->pos()->x(), dfstone->pos()->y());
-	myBodyDef.angle = -dfstone->rotation();
-	s->body = gameState.b2world->CreateBody(&myBodyDef);
-	b2CircleShape circleShape;
-	circleShape.m_radius = dfstone->radius();
-
-	b2FixtureDef fixtureDef;
-	fixtureDef.shape = &circleShape;
-	fixtureDef.density = 1;
-	s->body->CreateFixture(&fixtureDef);
-	s->body->SetUserData(s);
-}
-
-void initHidingSpot(HidingSpot *b, const FlatBuffGenerated::HidingSpot *dfhspot)
-{
-	b2BodyDef myBodyDef;
-	myBodyDef.type = b2_staticBody;
-	myBodyDef.position.Set(dfhspot->pos()->x(), dfhspot->pos()->y());
-	myBodyDef.angle = -dfhspot->rotation();
-	b->body = gameState.b2world->CreateBody(&myBodyDef);
-	b2CircleShape circleShape;
-	circleShape.m_radius = dfhspot->radius();
-
-	b2FixtureDef fixtureDef;
-	fixtureDef.shape = &circleShape;
-	fixtureDef.density = 1;
-	fixtureDef.filter.categoryBits = 0x0002;
-	b->body->CreateFixture(&fixtureDef);
-	b->body->SetUserData(b);
-}
-
 flatbuffers::Offset<FlatBuffGenerated::Level> serializeLevel(flatbuffers::FlatBufferBuilder &builder)
 {
+	//tileinfo
+	std::vector<flatbuffers::Offset<FlatBuffGenerated::Tileinfo>> tileinfoOffsets;
+	for (auto &ts : gameState.level->tileinfo) {
+		auto name = builder.CreateString(ts->name);
+		auto offset = FlatBuffGenerated::CreateTileinfo(builder, ts->gid, name);
+		tileinfoOffsets.push_back(offset);
+	}
+	auto tilesets = builder.CreateVector(tileinfoOffsets);
+
+	// objects
+	std::vector<flatbuffers::Offset<FlatBuffGenerated::Object>> objectOffsets;
+	for (auto &o : gameState.level->objects) {
+		FlatBuffGenerated::Vec2 pos(o->pos.x, o->pos.y);
+		auto hspotname = builder.CreateString(o->hspotname);
+		auto offset = FlatBuffGenerated::CreateObject(builder, &pos, o->rotation, o->gid, o->hspot, hspotname);
+		objectOffsets.push_back(offset);
+	}
+	auto objects = builder.CreateVector(objectOffsets);
+
+	// decoration
+	std::vector<flatbuffers::Offset<FlatBuffGenerated::Decoration>> decorationOffsets;
+	for (auto &d : gameState.level->decoration) {
+		FlatBuffGenerated::Vec2 pos(d->pos.x, d->pos.y);
+		auto offset = FlatBuffGenerated::CreateDecoration(builder, &pos, d->rotation, d->gid);
+		decorationOffsets.push_back(offset);
+	}
+	auto decoration = builder.CreateVector(decorationOffsets);
+
 	// hiding spots
 	std::vector<flatbuffers::Offset<FlatBuffGenerated::HidingSpot>> hspotOffsets;
-	for (auto &b : gameState.level->hidingspots)
+	for (auto &hs : gameState.level->hidingspots)
 	{
-		FlatBuffGenerated::Vec2 pos(b->body->GetPosition().x, b->body->GetPosition().y);
-		auto f = b->body->GetFixtureList();
-		auto c = (b2CircleShape *)f->GetShape();
-		auto off = FlatBuffGenerated::CreateHidingSpot(builder, c->m_radius, b->body->GetAngle(), &pos);
+		FlatBuffGenerated::Vec2 pos(hs->body->GetPosition().x, hs->body->GetPosition().y);
+		bool ellipse = false;
+		float radius = 0;
+		flatbuffers::Offset<flatbuffers::Vector<const FlatBuffGenerated::Vec2 *>> polyverts = 0;
```

Comment:
Why are you storing Vec2s here as pointers? The structures have only 2 floats in them, it adds complexity on account of having to use unique pointers

Reply:
That's what FlatBuffGenerated::CreateHidingSpot needed. But you have a point in that client doesn't need the hidingspots at all. Removed the section.